x/gov/keeper/proposal.go

```diff
@@ -184,9 +187,21 @@ func (keeper Keeper) SetProposalID(ctx sdk.Context, proposalID uint64) {
 	store.Set(types.ProposalIDKey, types.GetProposalIDBytes(proposalID))
 }
 
+func (keeper Keeper) GetVotingPeriod(ctx sdk.Context, content types.Content) time.Duration {
+	switch content.(type) {
+	case *types.TextProposal:
+		return keeper.GetVotingParams(ctx).VotingPeriodText
+	case *paramsproposal.ParameterChangeProposal:
+		return keeper.GetVotingParams(ctx).VotingPeriodParameterChange
+	case *upgradetypes.SoftwareUpgradeProposal, *upgradetypes.CancelSoftwareUpgradeProposal:
+		return keeper.GetVotingParams(ctx).VotingPeriodSoftwareUpgrade
+	}
+	return types.DefaultPeriod
```

Comment:
I didn't mean a hard-coded default btw, but an equally changeable parameter.
`DefaultVotingPeriod` or something.

Reply:
With which value ?

Reply:
With another parameter.

types.DefaultPeriod is just a hard-coded constant. No bueno imho

Reply:
Added a new parameter `VotingParams.VotingPeriodDefault` [ee2bded](https://github.com/atomone-hub/govgen/pull/16/commits/ee2bded6b8557a6180f423e5c9df66d5e0781211)

Reply:
Done here [ee2bded](https://github.com/atomone-hub/govgen/pull/16/commits/ee2bded6b8557a6180f423e5c9df66d5e0781211)